obci/interfaces/bci/analysis_master.py

```diff
@@ -0,0 +1,222 @@
+        """Create an empty buffer with given return function.
+        Needs to be implemented in subclass.
+
+        Args:
+            channel_count (int): number of channels in the input signal
+            ret_func: function which should be passed to a buffer as "ret_func"
+
+        Returns:
+            buffer instance
+        """
+        return None
+
+    @abc.abstractmethod
+    def create_classifier(self):
+        """Create a classifier to be used.
+        Needs to be implemented in subclass.
+
+        Returns:
+            AbstractClassifier. classifier instance
+        """
+        return None
+
+    @abc.abstractmethod
+    def identify_blink(self, blink):
+        """Identify to which target the given blink belongs, based on
+        some external (e.g. GUI) information. If this information
+        is not available (outside calibration phase), return None.
+        This method is NOT meant to use machine learning techniques.
+
+        Needs to be implemented in subclass.
+        This method will be run by a separate thread.
+
+        Args:
+            blink: information contained in a single blink
+
+        Returns:
+            string. name of the target
+            or None if identification could not be performed
+        """
+        return None
+
+    @abc.abstractmethod
+    def init_params(self):
+        """Needs to be implemented in subclass."""
+        pass
+
+    # -------------------------------
+    # optionally overridable methods
+    # -------------------------------
+
+    def classify(self, classifier, chunk, blink):
+        """Compute set of classification probabilities for a given blink.
+
+        May be re-implemented in subclass to perform some pre-processing
+        or feature extraction on the signal (chunk).
+        This method will be run by a separate thread.
+
+        Args:
+            classifier (AbstractClassifier): classifier instance to be used for classification
+            chunk: numpy 2D data array (channels × samples)
+            blink: information contained in a single blink
+
+        Returns:
+            dict. dictionary of {target: probability},
+            or None if classification could not be performed
+        """
+        return classifier.classify(chunk)
+
+    def learn(self, classifier, chunk, target):
+        """Learn that given chunk represents given target.
+
+        May be re-implemented in subclass to perform some pre-processing
+        or feature extraction on the signal (chunk).
+        This method will be run by a separate thread.
+
+        Args:
+            classifier (AbstractClassifier): classifier instance to be used for learning
+            chunk: numpy 2D data array (channels × samples)
+            target: name of the target
+        """
+        classifier.learn(chunk, target)
+
+    # ----------------
+    # internal methods
+    # ----------------
+
+    @log_crash
+    def __init__(self, addresses):
```

Comment:
children should be able to overwrite type.
For example type.ANALYSIS doesn't receive BLINKs only P300_ANALYSIS does.